### ISSUE TITLE
PATH does not support home relative paths

### ISSUE THREAD

**Issue by webdesserts** [nushell/nushell#4855]
### Describe the bug

In 0.59, it seems that the `PATH` environment variable does not support path expansion for `~/`. I'm not sure if this diverges from previous versions of nu, but this at least was unexpected for me when trying to migrate my configs over to the 0.59 beta.

### How to reproduce

1. Create a basic shell script at `~/bin/hello`
2. Inside Nu, run the following: `let-env PATH = ($env.PATH | append "~/bin")`
3. Try running `hello`, but receive "No such file or directory" error
4. Instead try using `path expand` before appending: `let-env PATH = ($env.PATH | append ("~/bin" | path expand))`
5. Try running `hello` again and the command succeeds

 

### Expected behavior

I would expect nu to automatically expand all paths in the `PATH` env variable so that you can easily expose bins stored inside your home directory.

### Screenshots

<img width="573" alt="Screen Shot 2022-03-15 at 3 32 09 PM" src="https://user-images.githubusercontent.com/916253/158467098-bcb78bb8-f802-4a84-a6af-80f9e3e08d79.png">

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.59.0                                   |
| branch             | main                                     |
| short_commit       | 42f1874a                                 |
| commit_hash        | 42f1874a3aae2e3bc3c669cb0aac0132a586ea24 |
| commit_date        | 2022-03-01 13:05:29 +00:00               |
| build_os           | macos-x86_64                             |
| rust_version       | rustc 1.59.0 (9d1b2106e 2022-02-23)      |
| rust_channel       | stable-x86_64-apple-darwin               |
| cargo_version      | cargo 1.59.0 (49d8809dc 2022-02-10)      |
| pkg_version        | 0.59.0                                   |
| build_time         | 2022-03-01 13:39:18 +00:00               |
| build_rust_channel | release                                  |
| features           | dataframe, default, trash, which, zip    |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
i'm curious about a couple of things.
1. if you do `$env.PATH` do you see your new path entry in there?
2. if you do `which hello` does it find your new executable?

**Comment by webdesserts** [comment]
> if you do `$env.PATH` do you see your new path entry in there?

You do see it in `$env.PATH` stored in its unexpanded form

> if you do `which hello` does it find your new executable?

`which hello` is a bit weird... it doesn't return anything for me even when it's successful.

<img width="569" alt="Screen Shot 2022-03-15 at 4 15 54 PM" src="https://user-images.githubusercontent.com/916253/158473990-b7a81d84-506e-48c7-8e73-8e524043798a.png">

**Comment by fdncred** [comment]
yup - sounds like you're right - nushell isn't expanding the `~` char here.

**Comment by uasi** [comment]
Just for the record, among zsh, bash and dash, only bash expands `~/` in PATH.

```
-> % zsh --version
zsh 5.8 (x86_64-apple-darwin19.6.0)

-> % zsh --no-rcs
% PATH="~/tmp:$PATH"
% echo "$PATH" | sed 's/:.*//'
~/tmp
% hello
zsh: command not found: hello
% ~/tmp/hello
Hello
```

```
-> % bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin20)
Copyright (C) 2007 Free Software Foundation, Inc.

-> % bash --norc --noprofile
bash-3.2$ PATH="~/tmp:$PATH"
bash-3.2$ echo "$PATH" | sed 's/:.*//'
~/tmp
bash-3.2$ hello
Hello
bash-3.2$ ~/tmp/hello
Hello
```

```
% dash # (version unknown; shipped with macOS 11.6.2)
$ PATH="~/tmp:$PATH"
$ echo "$PATH" | sed 's/:.*//'
~/tmp
$ hello
dash: 4: hello: not found
$ ~/tmp/hello
Hello
```

**Comment by webdesserts** [comment]
Yeah I was a little unsure on whether this was a bug or a feature request. One thing I will say is that from a configuration standpoint it’s very easy to fix the issue in nushell since you can just pass the resulting paths to `path expand`. It’s just that it’s a bit of a tripping hazard if you are expecting the path to work like it does in bash and it’s not immediately apparent why your scripts aren’t included.
